cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -180,7 +180,7 @@ func main() {
 		klog.Fatalf("Error getting server version: %v", err)
 	}
 
-	metricsManager := metrics.NewCSIMetricsManager("" /* driverName */)
+	metricsManager := metrics.NewCSIMetricsManagerWithOptions("" /* driverName */, metrics.WithMigration())
```

Comment:
Do we want to add the migration field for all drivers, or only those that are part of the migratable plugins? I believe this logic determines that: https://github.com/kubernetes-csi/external-provisioner/blob/37b54f06131df131cdcbe1084917075b5ea2a5a1/cmd/csi-provisioner/csi-provisioner.go#L332

Reply:
+1, yeah
I think we should only add to those that are part of migratable.

Reply:
I looked into it. Not sure if it is possible to set the migration field only for the migratable drivers here. 
https://github.com/kubernetes-csi/external-provisioner/blob/37b54f06131df131cdcbe1084917075b5ea2a5a1/cmd/csi-provisioner/csi-provisioner.go#L332
This logic requires a provisionername and we are getting the provisionerName after creating the gRPC client here : https://github.com/kubernetes-csi/external-provisioner/blob/37b54f06131df131cdcbe1084917075b5ea2a5a1/cmd/csi-provisioner/csi-provisioner.go#L185

And to create the gRPC client we need  to create the metricsManager.

Please let me know if you think there's some other way to do it.

Reply:
Hmm, I think there might be one possible way we can do.

We can create a new grpcClient with a new metricsManager after we get the provisionerName here:
https://github.com/kubernetes-csi/external-provisioner/blob/f937d0d4a1d88d866d95fad51148b686d9357cfc/cmd/csi-provisioner/csi-provisioner.go#L198

And move https://github.com/kubernetes-csi/external-provisioner/blob/f937d0d4a1d88d866d95fad51148b686d9357cfc/cmd/csi-provisioner/csi-provisioner.go#L329-L339 above so we know if this provisioner supports migration.

And dont forget to call grpcClient.Close() on the old connection to release resources.

Reply:
Addressed